pulsar-client-kafka-compat/pulsar-client-kafka/src/main/java/org/apache/kafka/clients/producer/PulsarKafkaProducer.java

```diff
@@ -244,13 +246,16 @@ public void close(long timeout, TimeUnit unit) {
         try {
             // Add the partitions info for the new topic
             cluster = cluster.withPartitions(readPartitionsInfo(topic));
-            List<org.apache.pulsar.client.api.ProducerInterceptor> wrappedInterceptors = interceptors.stream()
-                    .map(interceptor -> new KafkaProducerInterceptorWrapper(interceptor, keySerializer, valueSerializer, topic))
-                    .collect(Collectors.toList());
-            return pulsarProducerBuilder.clone()
-                    .topic(topic)
-                    .intercept(wrappedInterceptors.toArray(new org.apache.pulsar.client.api.ProducerInterceptor[wrappedInterceptors.size()]))
-                    .create();
+            if (keySerializer instanceof PulsarKafkaSchema && valueSerializer instanceof PulsarKafkaSchema) {
```

Comment:
why not change KafkaProducerInterceptorWrapper to use `Schema`? This would support all kind of schemas.

Reply:
I understand that this KafkaProducerInterceptorWrapper  class should be specially prepared for kafka, pulsar's own schema does not need to pass through this class, This judgment was added to handle the following logic:
```
List<org.apache.pulsar.client.api.ProducerInterceptor> wrappedInterceptors = interceptors.stream()
                        .map(interceptor -> new KafkaProducerInterceptorWrapper(interceptor, (PulsarKafkaSchema<K>) keySerializer, (PulsarKafkaSchema<V>) valueSerializer, topic))
                        .collect(Collectors.toList());
```

Reply:
> pulsar's own schema does not need to pass through this class, 

I don't understand why Pulsar's own schema doesn't need to pass through this class.